src/edge/models/fragment.py

```diff
@@ -91,8 +91,22 @@ def sorter_f(e):
 
             return out_edges[0].to_chunk
 
+    def lock(self):
+        """
+        Do a select for update, which, when executed inside a transaction, places
+        a lock on the fragment.
+        """
+
+        fragments = Fragment.objects.select_for_update().filter(pk=self.id)
+        # Lock only happens when querset is evaluated, therefore need to do at least fragments[0]
```

Comment:
Unless `querset` is a "short" name I'm not familiar with, I assume you meant `queryset` here. Not that it matters given that it's a comment...

Reply:
Yes, typo, will fix. Thanks.